addons/account/static/src/js/bills_upload.js

```diff
@@ -1,160 +1,170 @@
-            this.initialState.context['default_journal_id'] = kanbanRecord.id;
-            let journal_type = kanbanRecord.state.data.type;
-            if (journal_type == 'sale') {
-                this.initialState.context['default_move_type'] = 'out_invoice';
-            } else if (journal_type == 'purchase') {
-                this.initialState.context['default_move_type'] = 'in_invoice';
-            }
-        },
-
-        _onUpload: function (ev) {
-            this._setJournalContext(ev);
-            UploadDocumentMixin._onUpload.apply(this, arguments);
-        },
-
-        _onDrop: function (ev) {
-            this._setJournalContext(ev);
-            UploadDocumentMixin._onDrop.apply(this, arguments);
-        },
-    });
-
-    var DashboardKanbanView = KanbanView.extend({
-        config: Object.assign({}, KanbanView.prototype.config, {
-            Controller: DashboardKanbanController,
-        }),
-    });
-
-    viewRegistry.add('account_tree', DocumentsListView);
-    viewRegistry.add('account_documents_kanban', DocumentsKanbanView);
-    viewRegistry.add('account_dashboard_kanban', DashboardKanbanView);
-});
+    async onFileUploaded(file) {
+        let att_data = {
+            name: file.name,
+            mimetype: file.type,
+            datas: file.data,
+        }
+        const att_id = await this.orm.call("ir.attachment", "create", [att_data], {
+            context: {...this.props.extraContext, ...this.env.searchModel.context},
+        });
+        this.attachment_ids_to_process.push(att_id);
+    }
+
+    async onUploadComplete() {
+        const action = await this.orm.call("account.journal", "create_document_from_attachment", ["", this.attachment_ids_to_process], {
+            context: {...this.props.extraContext, ...this.env.searchModel.context},
+        });
+        this.attachment_ids_to_process = [];
+        this.action.doAction(action);
+    }
+}
+AccountFileUploader.components = {
+    FileUploader,
+}
+AccountFileUploader.template = "account.HiddenFileUploader";
+
+class AccountDropZone extends Component {}
+AccountDropZone.defaultProps = {
+    visibilityClass: "d-none",
+    dragOver: () => {},
+    dragLeave: () => {},
+    drop: () => {},
+}
+AccountDropZone.template = "account.DropZone";
+
+const AccountMoveDropzoneFunctions = {
+    setupDropZone() {
+        this.notificationService = useService("notification");
+        this.dzState = useState({
+            dzClass: "d-none",
+        })
+    },
+
+    showDropZone(ev) {
+        this.dzState.dzClass = "";
+    },
+
+    hideDropZone(ev) {
+        this.dzState.dzClass = "d-none";
+    },
+
+    onDrop(ev) {
+        const selector = '.account_file_uploader.o_input_file.o_hidden';
+        let uploadInput = ev.target.closest('.o_drop_area').querySelector(selector) || document.querySelector(selector);
+        let files = ev.dataTransfer ? ev.dataTransfer.files : false;
+        ev.preventDefault();
+        if (!!uploadInput && files) {
```

Comment:
why `!!uploadInput` instead of `uploadInput` ?

Reply:
[] is truthy - so bang bang

Reply:
Yes but what does `!!` change to that? If you do `Boolean([])` or `!![]`, you get the same result.

Reply:
Boolean(v) and !!v will produce the same result - just a matter of preference.

Reply:
`uploadInput` is not an array, maybe you meant to do `!!files`? But that would always be truthy anyways because as noted `[]` is truthy. Maybe you want `files.length > 0`?

`uploadInput` will be an `HTMLElement` or null. We also would not typically coerce values to booleans unless we have a good reason to do so (eg we want to call a function with the value in question and want to make sure we're not leaking object references)